src/services/formatting/formatting.ts

```diff
@@ -1117,6 +1117,25 @@ namespace ts.formatting {
         }
     }
 
+    export function getRangeOfEnclosingComment(sourceFile: SourceFile, position: number, onlyMultiLine: boolean): CommentRange | undefined {
+        const precedingToken = findPrecedingToken(position, sourceFile);
+        const trailingRangesOfPreviousToken = precedingToken && getTrailingCommentRanges(sourceFile.text, precedingToken.end);

```

Comment:
`findPrecedingToken` looks like it might be quite expensive. We should verify any impact on typing before checking this in - especially as some brace matching operations happen as a blocking operation on the UI thread.